src/sentry/search/events/builder.py

```diff
@@ -179,6 +475,541 @@ def get_snql_query(self) -> Query:
+
+        if combinator_name is not None and combinator is None:
+            raise InvalidSearchQuery(
+                f"{snql_function.name}: no support for the -{combinator_name} combinator"
+            )
+
+        if not snql_function.is_accessible(self.functions_acl, combinator):
+            raise InvalidSearchQuery(f"{snql_function.name}: no access to private function")
+
+        combinator_applied = False
+
+        arguments = snql_function.format_as_arguments(
+            name, parsed_arguments, self.params, combinator
+        )
+
+        self.function_alias_map[alias] = FunctionDetails(function, snql_function, arguments.copy())
+
+        for arg in snql_function.args:
+            if isinstance(arg, ColumnArg):
+                if (
+                    arguments[arg.name] in NumericColumn.numeric_array_columns
+                    and isinstance(arg, NumericColumn)
+                    and not isinstance(combinator, SnQLArrayCombinator)
+                ):
+                    arguments[arg.name] = Function(
+                        "arrayJoin", [self.resolve_column(arguments[arg.name])]
+                    )
+                else:
+                    arguments[arg.name] = self.resolve_column(arguments[arg.name])
+            if combinator is not None and combinator.is_applicable(arg.name):
+                arguments[arg.name] = combinator.apply(arguments[arg.name])
+                combinator_applied = True
+
+        if combinator and not combinator_applied:
+            raise InvalidSearchQuery("Invalid combinator: Arguments passed were incompatible")
+
+        if snql_function.snql_aggregate is not None:
+            if not resolve_only:
+                self.aggregates.append(snql_function.snql_aggregate(arguments, alias))
+            return snql_function.snql_aggregate(arguments, alias)
+
+        return snql_function.snql_column(arguments, alias)
+
+    def parse_function(self, match: Match[str]) -> Tuple[str, Optional[str], List[str], str]:
+        """Given a FUNCTION_PATTERN match, seperate the function name, arguments
+        and alias out
+        """
+        raw_function = match.group("function")
+        function, combinator = parse_combinator(raw_function)
+
+        if not self.is_function(function):
+            raise InvalidSearchQuery(f"{function} is not a valid function")
+
+        arguments = parse_arguments(function, match.group("columns"))
+        alias: Union[str, Any, None] = match.group("alias")
+
+        if alias is None:
+            alias = get_function_alias_with_columns(raw_function, arguments)
+
+        return (function, combinator, arguments, alias)
+
+    def get_public_alias(self, function: CurriedFunction) -> str:
+        """Given a function resolved by QueryBuilder, get the public alias of that function
+
+        ie. any_user_display -> any(user_display)
+        """
+        return self.function_alias_map[function.alias].field  # type: ignore
```

Comment:
```
src/sentry/search/events/builder.py:1011: error: Returning Any from function declared to return "str"
```
But I don't understand whyyy

Reply:
I think its cause `function.alias` might not exist, and `function_alias_map` might not have that key